samples/batch/transcription-enabled-storage/Connector/BatchClient.cs

```diff
@@ -0,0 +1,195 @@
+        }
+
+        public Task<TranscriptionFiles> GetTranscriptionFilesAsync(Guid id)
+        {
+            var path = $"{SpeechToTextBasePath}Transcriptions/{id}/files";
+            return this.GetAsync<TranscriptionFiles>(path);
+        }
+
+        public Task<Uri> PostTranscriptionAsync(string name, string description, string locale, Dictionary<string, string> properties, List<Uri> contentUrls, IEnumerable<Guid> modelIds)
+        {
+            var models = modelIds.Select(m => ModelIdentity.Create(m)).ToList();
+            var path = $"{SpeechToTextBasePath}Transcriptions/";
+
+            var transcriptionDefinition = TranscriptionDefinition.Create(name, description, locale, contentUrls, properties, models);
+            return this.PostAsJsonAsync(path, transcriptionDefinition);
+        }
+
+        public Task DeleteTranscriptionWithRetryAsync(Guid id)
+        {
+            var path = $"{SpeechToTextBasePath}Transcriptions/{id}";
+            return this.DeleteAsync(path);
+        }
+
+        private static Uri GetLocationFromPostResponseAsync(WebHeaderCollection headers)
+        {
+            return new Uri(headers["Location"]);
+        }
+
+        private static async Task<HttpWebResponse> GetHttpWebResponseAsync(HttpWebRequest request, ILogger logger)
+        {
+            var webresponse = await request.GetResponseAsync().ConfigureAwait(false);
+            var response = (HttpWebResponse)webresponse;
+
+            if (IsThrottledOrTimeoutStatusCode(response.StatusCode))
+            {
+                throw new TimeoutException();
+            }
+
+            if (response.StatusCode != HttpStatusCode.Accepted && response.StatusCode != HttpStatusCode.Created)
+            {
+                logger.LogInformation($"Failure: Status Code {response.StatusCode}, {response.StatusDescription}");
+                throw new NotImplementedException();
+            }
+
+            return response;
+        }
+
+        private HttpClient CreateHttpClient()
+        {
+            var client = new HttpClient();
+            client.Timeout = TimeSpan.FromMinutes(25);
+            client.BaseAddress = new Uri(HostName);
+            client.DefaultRequestHeaders.TransferEncodingChunked = null;
+            client.DefaultRequestHeaders.Add("Ocp-Apim-Subscription-Key", SubscriptionKey);
+            return client;
+        }
+
+        private async Task<Uri> PostAsJsonAsync(string path, TranscriptionDefinition payload)
+        {
+            var request = BuildWebRequest(path, payload);
+            Log.LogInformation("Request: " + request);
+            var webResponse = await GetHttpWebResponseAsync(request, Log).ConfigureAwait(false);
+            Log.LogInformation("StatusCode: " + webResponse.StatusCode);
+            return GetLocationFromPostResponseAsync(webResponse.Headers);
+        }
+
+        private HttpWebRequest BuildWebRequest(string path, TranscriptionDefinition payload)
```

Comment:
The name should be BuildPostWebRequest, since the request method only supports POST

Reply:
renamed